File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int>premp;
        int n=nums.size(),sum=0,subCount=0;
        premp[0]++;
        for(int i=0;i<n;i++)
        {
            sum+=nums[i];
            if(premp.count(sum-k))
                subCount+=premp[sum-k];
            premp[sum]++;
        }
        return subCount;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer,Integer> mp = new HashMap<>();
        int sum=0,res=0;
        mp.put(0,1);
        for(int i=0;i<nums.length;i++)
        {
            sum+=nums[i];
            res+=mp.getOrDefault(sum-k,0);
            mp.put(sum, mp.getOrDefault(sum, 0) + 1);
        }
        return res;
    }
}